cmake/toolchains/ARM.cmake

```diff
@@ -42,14 +42,10 @@ function(mbed_set_toolchain_options target)
             $<$<COMPILE_LANGUAGE:CXX>:${common_options}>
     )
 
-    set(asm_preproc_options
-        "--target=arm-arm-none-eabi,-D,MBED_CONF_PLATFORM_CRASH_CAPTURE_ENABLED"
-    )
     target_compile_options(${target}
         PUBLIC
+            $<$<COMPILE_LANGUAGE:ASM>:--target=arm-arm-none-eabi -masm=armasm>
```

Comment:
`-masm=auto` works for me. If confirmed, I'll change this

Reply:
`-masm=auto` also worked on my Mac OS system with `cmake version 3.17.3`.
However, would it not be prudent to explicitly set it?
```
In rare circumstances, the auto-detection might select the wrong assembler for the input file.
For example, if the input file is a .S file that requires preprocessing, auto-detection might select
the wrong assembler. For the files where auto-detection selects the wrong assembler, you must
select -masm=gnu or -masm=armasm explicitly.
```
Source: https://www.keil.com/support/man/docs/armclang_ref/armclang_ref_yml1538053827173.htm

Reply:
For now, armasm would be better but what about the future? New MCU startup files will soon have GNU syntax for ARMClang .S files, therefore this armasm won't work. Auto mode is neat and exactly what we need. if we find bugs, we shall report it to the compiler (it is state as "rare").